### PR TITLE
Avoid manually setting up in-process GL bindings for Skia

### PR DESCRIPTION
**Pull request by chinmaygarde**
In a subsequent commit, I will get rid of the python script and its generated files.

### CODE REVIEW

File: skia/BUILD.gn [hunk 1]
```diff
@@ -368,6 +368,30 @@ component("skia") {
     "//third_party/skia/src/fonts/SkGScalerContext.h",
   ]
 
+  if (is_ios || is_mac || is_android) {
+    sources -= [
+      "//third_party/skia/src/gpu/gl/GrGLCreateNativeInterface_none.cpp"
+    ]
+
+    set_sources_assignment_filter([])
```
Comment (eseidelGoogle) [inline skia/BUILD.gn]: Is this necessary?  Isn't this filter supposed to handle this for us?

File: skia/BUILD.gn [hunk 2]
```diff
@@ -368,6 +368,30 @@ component("skia") {
...
+
+    if (is_ios) {
+      sources += [
```
Comment (eseidelGoogle) [inline skia/BUILD.gn]: If the filter is working then we could just list all 3 of these in one unified sources block.

File: sky/shell/BUILD.gn [hunk 3]
```diff
@@ -96,6 +96,10 @@ if (is_android) {
              ":jni_headers",
              ":common",
            ]
+    ldflags = [
```
Comment (eseidelGoogle) [inline sky/shell/BUILD.gn]: I'm not sure if this is the right place to put these flags, but seems reasonable.

**Comment by eseidelGoogle** [comment]
@abarth is your best reviewer, but lgtm.  Please wait for his look.

**Comment by abarth** [comment]
LGTM

File: skia/BUILD.gn (hunk 1 above)
Comment (chinmaygarde) [inline skia/BUILD.gn]: The iOS in the name is oddly capitalized for the filter.

**Pull request merged**